modules/programs/i3lock.nix

```diff
@@ -0,0 +1,412 @@
+          default = cfg.text.size.default;
+        };
+
+        wrong = mkOption {
+          description = "i3lock text size for displaying wrong text.";
+          type = types.nullOr types.int;
+          default = cfg.text.size.default;
+        };
+
+        greeter = mkOption {
+          description = "i3lock text size for displaying greeter text.";
+          type = types.nullOr types.int;
+          default = cfg.text.size.default;
+        };
+
+        default = mkOption {
+          description = "i3lock default text size";
+          type = types.nullOr types.int;
+          default = null;
+        };
+      };
+    };
+
+    colors = {
+      insideVerify = mkOption {
+        description = "i3lock inside verify color";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      ringVerify = mkOption {
+        description = "i3lock ring verify color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      insideWrong = mkOption {
+        description = "3lock inside wrong color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      ringWrong = mkOption {
+        description = "i3lock ring wrong color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      inside = mkOption {
+        description = "3lock inside color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      ring = mkOption {
+        description = "i3lock ring color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      line = mkOption {
+        description = "i3lock line color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      separator = mkOption {
+        description = "i3lock seprarator color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      verify = mkOption {
+        description = "i3lock verify color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      wrong = mkOption {
+        description = "i3lock wrong color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      time = mkOption {
+        description = "i3lock time color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      date = mkOption {
+        description = "i3lock date color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      layout = mkOption {
+        description = "i3lock layout color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      keyhl = mkOption {
+        description = "i3lock key hold color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      bshl = mkOption {
+        description = "i3lock backspace hold color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+    };
+
+    extraCommands = mkOption {
```

Comment:
Do you think it makes sense to have two options: `extraCommandsBefore` and `extraCommandsAfter`? E.g. I've been using the following locking script to make sure there is only US layout when i3lock is working and reseting it to the default configuration after successful unlock:

```
  locker = pkgs.writeShellScriptBin "lock" ''
    ${pkgs.xorg.setxkbmap}/bin/setxkbmap us
    ${pkgs.i3lock-color}/bin/i3lock-color --nofork
    ${pkgs.systemd}/bin/systemctl --user restart setxkbmap.service
  '';
```
Note `--nofork` otion that should be used in this case.

If you think this is too special case to cover in this module, we can leave it as is :)

Reply:
Makes sense, added suggested `extraCommandBefore` and `extraCommandsAfter`.